### PR TITLE
feat(core): add core skeleton for solana

### PR DESCRIPTION
**Pull request by alebusse**
Added core skeleton for solana   
Uncommented accounts for solana in account-lib coins

[stlx-8776](https://bitgoinc.atlassian.net/browse/STLX-8776)

### CODE REVIEW

**Review by joey-bai — APPROVED** [review]

**Review by CapnMigraine — APPROVED** [review]

File: modules/core/src/v2/coins/sol.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  txPrebuild: TransactionPrebuild;
+  prv: string | string[];
+  pubKeys?: string[];
+}
+export interface TransactionPrebuild extends BaseTransactionPrebuild {
+  txHex: string;
+  txInfo: TxInfo;
+  source: string;
+}
+
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/sol.ts:54]: May be we need a VerifiedTransactionParameters interface.

File: modules/core/src/v2/coins/sol.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  getFamily(): CoinFamily {
+    return this._staticsCoin.family;
+  }
+  getFullName(): string {
+    return this._staticsCoin.fullName;
+  }
+  getBaseFactor(): string | number {
+    return Math.pow(10, this._staticsCoin.decimalPlaces);
+  }
+
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/sol.ts:84]: Can we do in solana transfers with amount zero?

File: modules/core/src/v2/coins/sol.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  ): Bluebird<SignedTransaction> {
+    throw new MethodNotImplementedError('signTransaction method not implemented');
+  }
+
+  parseTransaction(params: any, callback?: NodeCallback<any>): Bluebird<any> {
+    throw new MethodNotImplementedError('parseTransaction method not implemented');
+  }
+
+  /**
+   * Explain a Stacks transaction from txHex
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/sol.ts:144]: Modify Ts doc

File: modules/core/src/v2/coins/sol.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+   * @param params
+   * @param callback
+   */
+  explainTransaction(
+    params: ExplainTransactionOptions,
+    callback?: NodeCallback<StxTransactionExplanation>
+  ): Bluebird<StxTransactionExplanation> {
```
Comment (rliberal-simtlix) [inline modules/core/src/v2/coins/sol.ts:151]: change stx for Sol

File: modules/core/test/v2/unit/coins/sol.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  it('should instantiate the coin', function () {
+    let localBasecoin = bitgo.coin('tsol');
+    localBasecoin.should.be.an.instanceof(Tsol);
+
+    localBasecoin = bitgo.coin('sol');
+    localBasecoin.should.be.an.instanceof(Sol);
+  });
+
+  it('should retun the right info', function() {
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:23]: should return

File: modules/core/test/v2/unit/coins/sol.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    basecoin.getChain().should.equal('tsol');
+    basecoin.getFamily().should.equal('sol');
+    basecoin.getFullName().should.equal('Testnet Sol');
+    basecoin.getBaseFactor().should.equal(1000000000);
+  });
+
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:29]: Add todo to complete tests

**Pull request merged**